### PR TITLE
CARTO: Handle minZoom&maxZoom in H3TileLayer

### PR DESCRIPTION
**Pull request by felixpalmer**
For https://github.com/visgl/deck.gl/issues/6912

#### Change List
- Use `minZoom`&`maxZoom` props in addition to `min/maxresolution` when limiting H3 tile requests

### CODE REVIEW

File: modules/carto/src/layers/h3-tile-layer.ts [hunk 1]
```diff
@@ -65,15 +65,35 @@ export default class H3TileLayer<DataT = any, ExtraPropsT = {}> extends Composit
...
+    // of a Tileset2D refer to the resolution levels, not the Mercator zooms
     return [
       new SpatialIndexTileLayer(this.props, {
         id: `h3-tile-layer-${this.props.id}`,
         data,
         // @ts-expect-error Tileset2D should be generic over TileIndex
         TilesetClass: H3Tileset2D,
         renderSubLayers,
-        maxZoom
+        minZoom: minresolution,
```
Comment (alasarr) [inline modules/carto/src/layers/h3-tile-layer.ts:95]:
```suggestion
        // minZoom and maxZoom are in h3 resolution, they will be converted to XYZ zoom levels by H3Tileset2D
        minZoom: minresolution,
```

**Review by alasarr — APPROVED** [review]
LGTM, just added a commit suggestion with a comment to make it easier to understand

File: modules/carto/src/layers/h3-tile-layer.ts (hunk 1 above)
Comment (felixpalmer) [reply modules/carto/src/layers/h3-tile-layer.ts:95]: That's not quite right, H3Tileset2D will treat them as H3 resolutions, it is just the base Tileset2D class is hardcoded to treat zoom and resolution as equal

Comment (felixpalmer) [inline modules/carto/src/layers/h3-tile-layer.ts:95]:
```suggestion
        // minZoom and maxZoom are H3 resolutions, however we must use this naming as that is what the Tileset2D class expects
        minZoom: minresolution,
```

Comment (alasarr) [reply modules/carto/src/layers/h3-tile-layer.ts:96]: ok, good to know :)

**Pull request merged**